File: programs/00_MyInsertionSort/myInsertionSort.c
#include <stdio.h>
void myInsertionSort(int arr[],int n);
int main(int argc, char* argv){
	int n;
	scanf("%d",&n);
	int a[n];
	for(int i=0;i<n;i++){
		scanf("%d",&a[i]);
	}
	myInsertionSort(a,n);
	for(int i=0;i<n;i++){
		printf("%d\t",a[i]);
	}
	printf("\n");
	return 0;
}
void myInsertionSort(int arr[],int n){
	int i,j,temp;
	for(int i=0;i<n;i++){
		for(j=i;j-1>=0;j--){
			if(arr[j]<arr[j-1]){
				temp = arr[j];
				arr[j] = arr[j-1];
				arr[j-1]= temp;
			}
			else{
				break;
			}
		}
	}
}

File: programs/00_MyInsertionSort/myInsertionSort.py
def myInsertionSort(arr, n):
	for i in range(0,n):
		j = i
		while(j-1>=0):
			if arr[j]<arr[j-1]:
				arr[j], arr[j-1] = arr[j-1], arr[j]
			else:
				break
			j = j - 1
def main():
	print("No of elements")
	x = int(input());
	arr = []
	for i in range(0,x):
		arr.append(int(input()))
	myInsertionSort(arr,len(arr))
	for i in arr:
		print(str(i)+"\t",end="")
	print()
if __name__ == '__main__':
	main()
